torch/distributed/_sharded_optim/__init__.py

```diff
@@ -0,0 +1,61 @@
+from typing import Iterator, Tuple, Union
+from .api import ShardedOptimizer
+
+import torch.nn as nn
+
+from torch.distributed._sharded_tensor import (
+    ShardedTensor
+)
+
+def module_named_params_with_sharded_tensor(
+    module: nn.Module,
+    prefix: str = '',
+    recurse: bool = True,
+) -> Iterator[Tuple[str, Union[nn.Parameter, ShardedTensor]]]:
+
+    r"""Returns an iterator over module parameters (together with the
+    ShardedTensor parameters), yielding both the name of the parameter
+    as well as the parameter itself. This is typically passed to an
+    ShardedOptimizer
```

Comment:
You probably want to use :class:`torch.distributed._sharded_optim.ShardedOptimizer` or just ``ShardedOptimizer`` here.